esmcol_validator/validator.py

```diff
@@ -0,0 +1,359 @@
+                valid_dir = True
+
+            except Exception:
+                logger.exception('ESMCol Download Error')
+                raise VersionException(f'Could not download ESMCol specification')
+        else:
+            valid_dir = False
+            for esmcol_spec_dir in self.esmcol_spec_dirs:
+                spec_file = Path(esmcol_spec_dir) / spec_name / '.json'
+                if spec_file.is_file():
+                    valid_dir = True
+                    try:
+                        logging.debug('Gather ESMCol specs from local directory.')
+                        with open(spec_file, 'r') as f:
+                            spec = json.load(f)
+
+                    except FileNotFoundError:
+                        try:
+                            logger.critical(
+                                f'Houston, we have a problem! Could not find spec file {spec_file}'
+                            )
+                            url = getattr(Esmcol, f'{spec_name}_schema_url')
+                            spec = requests.get(url(self.esmcat_version)).json()
+
+                        except:
+                            logger.exception(
+                                'The ESMCol specification file does not exist or does not match the ESMCol file you are trying '
+                                'to validate. Please check your esmcol_spec_dirs path.'
+                            )
+                            sys.exit(1)
+                    except Exception as e:
+                        logging.exception(e)
+
+        if valid_dir:
+            file_name = (
+                Path(self.dirpath) / f"{spec_name}_{self.esmcat_version.replace('.','_')}.json"
+            )
+            with open(file_name, 'w') as fp:
+                logging.debug(f'Copying {spec_name} spec from local file to cache')
+                fp.write(json.dumps(spec))
+        else:
+            logger.exception(
+                'The ESMCol specification file does not exist or does not match the ESMCol file you are trying '
+                'to validate. Please check your esmcol_spec_dirs path.'
+            )
+            sys.exit(1)
+
+        return spec
+
+    def validate_json(self, content, schema):
+        """ Validate ESMCol
+        Parameters
+        ----------
+        content : dict
+             input ESMCol file content
+        schema : dict
+             schema of ESMCol
+
+        Returns
+        -------
+        validation message
+        """
+        try:
+            logging.info('Validating ESMCol')
+            validate(content, schema)
+            return True, None
+        except ValidationError as e:
+            logger.warning('ESMCol Validation Error')
+            return False, f'{e.message} of {list(e.path)}'
+
+        except Exception as e:
+            logger.exception('ESMCol error')
+            return False, f'{e}'
+
+    def validate_csv(self, content):
```

Comment:
> Do you check the csv file for consistency with the json file?

Yes. This is where the csv file is happening. I may be missing some edge cases here, but the more general cases are well covered (I think :))